pandas/core/reshape/pivot.py

```diff
@@ -427,6 +427,9 @@ def _convert_by(by):
 @Substitution("\ndata : DataFrame")
 @Appender(_shared_docs["pivot"], indents=1)
 def pivot(data: "DataFrame", index=None, columns=None, values=None) -> "DataFrame":
+    if columns is None:
+        raise ValueError("`columns` is not optional.")
```

Comment:
since I don't think we can change the function signature, then to be semi consistent with standard error messages, then maybe `TypeError: pivot() missing 1 required argument: 'columns'`

(or an enhancement to the code so that columns is the remaining columns not in index and values arguments, and then raise a message that at least one of columns and values must be specified